python/tests/catalog/test_catalog_base.py

```diff
@@ -0,0 +1,131 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import pytest
+
+from iceberg.catalog import base
+
+
+def test_namespace_init():
+    namespace = base.Namespace("foo", "bar", "baz")
+
+    assert namespace.levels == ("foo", "bar", "baz")
+    assert namespace.empty() is False
+    assert namespace.level(0) == "foo"
+    assert namespace.level(1) == "bar"
+    assert namespace.level(2) == "baz"
+    assert hash(namespace) == 1445470134
+    assert len(namespace) == 3
+    assert str(namespace) == "foo.bar.baz"
```

Comment:
nit: shall we have one for `assert repr(namespace)` as well? 

I do see there's `test_namespace_repr ` below but we also had`test_namespace_hash` and we tested hash here :)